frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/actions/option_action.dart

```diff
@@ -30,6 +48,98 @@ class DividerOptionAction extends CustomActionCell {
   }
 }
 
+class AlignOptionAction extends PopoverActionCell {
+  AlignOptionAction({
+    required this.editorState,
+  });
+
+  final EditorState editorState;
+
+  @override
+  Widget? leftIcon(Color iconColor) {
+    return FlowySvg(
+      name: 'editor/align/$align',
+      size: const Size.square(12),
+    ).padding(all: 2.0);
+  }
+
+  @override
+  String get name {
+    return LocaleKeys.document_plugins_optionAction_align.tr();
+  }
+
+  @override
+  Widget Function(
+    BuildContext context,
+    PopoverController parentController,
+    PopoverController controller,
+  ) get builder => (context, parentController, controller) {
+        final selection = editorState.selection?.normalized;
+        if (selection == null) {
+          return const SizedBox.shrink();
+        }
+        final node = editorState.getNodeAtPath(selection.start.path);
+        if (node == null) {
+          return const SizedBox.shrink();
+        }
+        final List<Widget> children =
+            OptionAlignType.values.map((e) => OptionAlignWrapper(e)).map((e) {
+          final leftIcon = e.leftIcon(Theme.of(context).colorScheme.onSurface);
+          final rightIcon =
+              e.rightIcon(Theme.of(context).colorScheme.onSurface);
+          return HoverButton(
+            onTap: () async {
+              await onAlignChanged(e.inner);
+              controller.close();
+              parentController.close();
+            },
+            itemHeight: ActionListSizes.itemHeight,
+            leftIcon: leftIcon,
+            name: e.name,
+            rightIcon: rightIcon,
+          );
+        }).toList();
+
+        return IntrinsicHeight(
+          child: IntrinsicWidth(
+            child: Column(
+              children: children,
+            ),
+          ),
+        );
+      };
+
+  String get align {
+    final selection = editorState.selection;
+    if (selection == null) {
+      return 'center';
+    }
+    final node = editorState.getNodeAtPath(selection.start.path);
+    final align = node?.attributes['align'];
+    return align ?? 'center';
+  }
+
+  Future<void> onAlignChanged(OptionAlignType align) async {
+    final name = align.name;
+    if (name == this.align) {
+      return;
+    }
+    final selection = editorState.selection;
+    if (selection == null) {
+      return;
+    }
+    final node = editorState.getNodeAtPath(selection.start.path);
```

Comment:
Merely a thought I have, since `getNodeAtPath` further down supports taking an empty list, we might as well make editorState.getNodeAtPath able to take a null value, and if it's null just return it.

That way we can simplify these cases from
```
final selection = editorState.selection;
    if (selection == null) {
      return;
    }
    final node = editorState.getNodeAtPath(selection.start.path);
    if (node == null) {
      return;
    }
```
To
```
    final selection = editorState.selection;
    final node = editorState.getNodeAtPath(selection?.start.path);
    if (selection == null || node == null) {
        return;
    }
```And since we do this a lot, I think it'll be better for readability in the long run.

Reply:
Agree. I will modify this logic later, not on this PR